Name: Abhilasha Ojha

Phone number: 7003785194

Email address: abhilasha.ojha1998@gmail.com

Linkedin profile: https://www.linkedin.com/in/abhilasha-ojha-a59330171/

Github profile: https://github.com/abhilashaojha

Analyzing an affiliated credit card (special card only available to members of a specific group). Think of it like this- the credit card company is like Citibank and the affiliated group is a company like Indian Oil or Barista. The affiliated group has to pay some fees to credit card company (because the credit card company is providing this service to the group)

In [1]:
#importing neccessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Consider parameters for the above problem:

    Average card balance is Rs 1000.
    Each card has an interest rate of 15%
    Membership fee is Rs 20 per card
    Loss rate is 3%
    Rs 25 operating cost per card
    Rs 10 affiliation fee per card (cost to the group organization itself, so that their members get this awesome card program)
    6.5% cost of funds (that credit card company must pay to the bank where it gets money)
    Bill/statement is issued on 1st of every month
    Credit-free period = 15 days from the bill/statement issue date.

Considering our timeline starts from 1st of January,2019. We calculate the revenue,Net profit/loss by the end of the year,that is, December,2019.

In [2]:
#initializing the parameters on monthly basis

avg_bal=1000 #average card balance
rate=15/100 #interest rate of 15% per annum
mem_fee=20*12 #assuming Membership fee monthly---converting it annually
loss_rate=3/100 #loss rate
opr_cost=25*12 #assuming operating cost per card monthly---converting it annually.
affl_card=10*12 #assuming affiliation cost per card monthly---converting it annually.
cof_r=6.5/100 #Cost of funds rate
credit_free=15 #credit-free period

Consider 1000 customers for the above scenario, divide these 1000 customers into following groups(upon your assumption and understanding about the credit lending industry)

>Customer who pay bill within the credit-free period

>Customers paying 30 days after the due-date.

>Customer paying 45 days after the last due date.

>Customer paying 60 days after the last due date (remove these customers for the complete calculation for the next month, basically you are not supposed to provide further credit services to these customer)

Assumptions made on the basis of number of customers, with the aforementioned conditions are stated below:

The customers who pay bill within credit-free period are **'90%'** of the total customers in the database.

The customers paying 30 days after the last due-date are **'5%'** of the total customers in the database.

The customer paying 45 days after the last due date are **'3%'** of the total customers in the database.

The customer paying 60 days after the last due date are **'2%'** of the total customers in the database.

In [3]:
#Let's initialize the parameters
active_cust=1000  #Customers who are present in the database after the removal of the customers who are paying the credit-bill 60 days after the due-date.
cust=1000 #total customers

def cust_update(active_cust):
    #let the customers who pay bill within credit-free period are 90%
    global cust_no_delay
    cust_no_delay=round(active_cust*(90/100))

    #Let the customers paying 30 days after the due-date are 5%
    global cust_30_delay
    cust_30_delay=round(active_cust*(5/100))

    #Let the customer paying 45 days after the last due date are 3%
    global cust_45_delay
    cust_45_delay=round(active_cust*(3/100))

    #Let the customer paying 60 days after the last due date are 2%
    global cust_60_delay
    cust_60_delay=round(active_cust*(2/100))
    return active_cust


In [4]:
cust_update(active_cust)

1000

Category-wise distribution of customers are done with the help of list **'category_type'** and **'Category_cust'**. 

In [5]:
category_type=[cust_no_delay,cust_30_delay,cust_45_delay,cust_60_delay]
Category_cust=['Paid with no delay','Paid after 30 days','Paid after 45 days','Paid after 60 days']

Calculation of interest after 'n' days of payement:
$$
Interest(n) = (n * OutstandingBalance * InterestRatePerYear)/365
$$

**Calculation of Interest**

A user-defined function **'interest_amount(n)'** is made which calculates the Interest for each customer category where n is customer category. 

In [6]:
#Billing cycle starts from 1st of every month
#Calculating monthly interest

def interest_amount(n):
    if n==category_type[0]: #customers paying credit card bill within due date
        interest=0
    elif n==category_type[1]:  #customers paying credit card bill 30 days after last due date
        interest=(((credit_free + 30)*avg_bal*rate)/365)*category_type[1]
    elif n==category_type[2]:   #customers paying credit card bill 45 days after last due date
        interest=(((credit_free + 45)*avg_bal*rate)/365)*category_type[2]
    else:
        interest=(((credit_free + 60)*avg_bal*rate)/365)*category_type[3]  #customers paying credit card bill 60 days after last due date
        #Remove these customers from the credit-service
    return interest 

A function **'calculations()'** is described which is responsible for the calculation of the interest, Principal Amount and total Amount. All the parameters are list which are getting appended with every iteration till the 4th iteration. 


In [7]:
#Categorywise calculation of Interest and other parameters:

def calculations():
   global interest
   interest=[] #stores the interest of the active customers
   principal=[] #stores the principal amount of the active customers
   global Amount
   Amount=[] #stores the overall amount(principal + interest) of the active customers
   Amount.clear()
   principal.clear()
   interest.clear()
   for i in range(4):
      Amount.append(category_type[i]*avg_bal + interest_amount(category_type[i]))
      principal.append(category_type[i]*avg_bal)
      interest.append(interest_amount(category_type[i]))
   return Amount


Creating an empty dataframe, **'data_new'** with the column names *'Category_cust','Number of customers','Amount','Interest','month'*

This dataframe will be further used to concatenate temporary dataframes.

In [8]:
data_new=pd.DataFrame(columns={'Category_cust','Number of customers','Amount','Interest','month'})
data_new

Empty DataFrame
Columns: [month, Interest, Amount, Number of customers, Category_cust]
Index: []

**Asuumption:**

* Now, according to our timeline, the first bill/statement would be issued on 1st of February,2019 (since we are considering January,2019 to be the day of first transaction). This implies that till 15th february would be the credit-free period for customers to pay their bills. 

* By the second week of April(12th April,2019), we would have the our list of customers falling in different categories mentioned above. We will revoke credit services of the customers falling in the 4th category(defaulters) from the list of customers and thereby our dataframe would be updated accordingly.

* This is how the below piece of code works for every cycle after April. Hence the 'month' column in the dataframe starts with the number 4, that means the 4th month, i.e, April.

In [9]:
for i in range(9):
    print("Active customers in the dataframe:",active_cust)
    cust_update(active_cust)
    category_type=[cust_no_delay,cust_30_delay,cust_45_delay,cust_60_delay]
    calculations()
    data_temp=pd.DataFrame({'Category_cust': Category_cust,'Number of customers':category_type,'Amount': Amount,'Interest':interest,'month':i+4})
    data_new=pd.concat([data_new,data_temp])
    active_cust=active_cust - cust_60_delay
    cust_update(active_cust)
data_new

Active customers in the dataframe: 1000
Active customers in the dataframe: 980
Active customers in the dataframe: 960
Active customers in the dataframe: 941
Active customers in the dataframe: 922
Active customers in the dataframe: 904
Active customers in the dataframe: 886
Active customers in the dataframe: 868
Active customers in the dataframe: 851


month    Interest        Amount Number of customers       Category_cust
0     4         0.0      900000.0                 900  Paid with no delay
1     4  924.657534  50924.657534                  50  Paid after 30 days
2     4  739.726027  30739.726027                  30  Paid after 45 days
3     4  616.438356  20616.438356                  20  Paid after 60 days
0     5         0.0      882000.0                 882  Paid with no delay
1     5  906.164384  49906.164384                  49  Paid after 30 days
2     5  715.068493  29715.068493                  29  Paid after 45 days
3     5  616.438356  20616.438356                  20  Paid after 60 days
0     6         0.0      864000.0                 864  Paid with no delay
1     6  887.671233  48887.671233                  48  Paid after 30 days
2     6  715.068493  29715.068493                  29  Paid after 45 days
3     6  585.616438  19585.616438                  19  Paid after 60 days
0     7         0.0      847000.0                 847  Paid with no delay
1     7  869.178082  47869.178082                  47  Paid after 30 days
2     7  690.410959  28690.410959                  28  Paid after 45 days
3     7  585.616438  19585.616438                  19  Paid after 60 days
0     8         0.0      830000.0                 830  Paid with no delay
1     8  850.684932  46850.684932                  46  Paid after 30 days
2     8  690.410959  28690.410959                  28  Paid after 45 days
3     8  554.794521  18554.794521                  18  Paid after 60 days
0     9         0.0      814000.0                 814  Paid with no delay
1     9  832.191781  45832.191781                  45  Paid after 30 days
2     9  665.753425  27665.753425                  27  Paid after 45 days
3     9  554.794521  18554.794521                  18  Paid after 60 days
0    10         0.0      797000.0                 797  Paid with no delay
1    10   813.69863   44813.69863                  44  Paid after 30 days
2    10  665.753425  27665.753425                  27  Paid after 45 days
3    10  554.794521  18554.794521                  18  Paid after 60 days
0    11         0.0      781000.0                 781  Paid with no delay
1    11  795.205479  43795.205479                  43  Paid after 30 days
2    11   641.09589   26641.09589                  26  Paid after 45 days
3    11  523.972603  17523.972603                  17  Paid after 60 days
0    12         0.0      766000.0                 766  Paid with no delay
1    12  795.205479  43795.205479                  43  Paid after 30 days
2    12   641.09589   26641.09589                  26  Paid after 45 days
3    12  523.972603  17523.972603                  17  Paid after 60 days

Let us answer the following questions: 

**1. How much profit % is generated in 1 year?**

In [10]:
#calculating profit at the end of the year
profit=1000*(mem_fee + affl_card) + sum(data_new['Interest'])
print(profit)

378955.4794520548


In [11]:
#Liability calculation
liability=1000*((1000*cof_r) + opr_cost + 1000*loss_rate) #Assuming the loss rate on the outstanding balance default.
print(liability)

395000.0


* As we can see the company suffers a loss of $16,045 (Liability - profit) at the end of the year. There is no profit at the end of the year. 

**2. The 3% loss rate can either mean 3% of the outstanding balance defaults or 3% of the borrowers default on their entire balance. If the average balance is increased to $2000, what happens to the loss? What happens to the loss?**

* The Liability is calculated considering the loss rate on the outstanding balance default. The amount that the company has to pay back at the end of the year is $3,95,000 and the amount that the company gets at the end of the year is $3,78,955.480(rounding off),that gives a total loss of $16,045.

* If the average balance is increased to $2000 ,then keeping the numbers of customers same, the following scenarios takes place:


In [12]:
data_new1=pd.DataFrame(columns={'Category_cust','Number of customers','Amount','Interest','month'})
data_new1

Empty DataFrame
Columns: [month, Interest, Amount, Number of customers, Category_cust]
Index: []

In [13]:
avg_bal=2000
active_cust=1000
cust=1000
for i in range(9):
    print("Active customers in the dataframe:",active_cust)
    cust_update(active_cust)
    category_type=[cust_no_delay,cust_30_delay,cust_45_delay,cust_60_delay]
    print(category_type)
    calculations()
    data_temp1=pd.DataFrame({'Category_cust': Category_cust,'Number of customers':category_type,'Amount': Amount,'Interest':interest,'month':i+4})
    data_new1=pd.concat([data_new1,data_temp1])
    active_cust=active_cust - cust_60_delay
    cust_update(active_cust)
data_new1

Active customers in the dataframe: 1000
[900, 50, 30, 20]
Active customers in the dataframe: 980
[882, 49, 29, 20]
Active customers in the dataframe: 960
[864, 48, 29, 19]
Active customers in the dataframe: 941
[847, 47, 28, 19]
Active customers in the dataframe: 922
[830, 46, 28, 18]
Active customers in the dataframe: 904
[814, 45, 27, 18]
Active customers in the dataframe: 886
[797, 44, 27, 18]
Active customers in the dataframe: 868
[781, 43, 26, 17]
Active customers in the dataframe: 851
[766, 43, 26, 17]


month     Interest         Amount Number of customers       Category_cust
0     4          0.0      1800000.0                 900  Paid with no delay
1     4  1849.315068  101849.315068                  50  Paid after 30 days
2     4  1479.452055   61479.452055                  30  Paid after 45 days
3     4  1232.876712   41232.876712                  20  Paid after 60 days
0     5          0.0      1764000.0                 882  Paid with no delay
1     5  1812.328767   99812.328767                  49  Paid after 30 days
2     5  1430.136986   59430.136986                  29  Paid after 45 days
3     5  1232.876712   41232.876712                  20  Paid after 60 days
0     6          0.0      1728000.0                 864  Paid with no delay
1     6  1775.342466   97775.342466                  48  Paid after 30 days
2     6  1430.136986   59430.136986                  29  Paid after 45 days
3     6  1171.232877   39171.232877                  19  Paid after 60 days
0     7          0.0      1694000.0                 847  Paid with no delay
1     7  1738.356164   95738.356164                  47  Paid after 30 days
2     7  1380.821918   57380.821918                  28  Paid after 45 days
3     7  1171.232877   39171.232877                  19  Paid after 60 days
0     8          0.0      1660000.0                 830  Paid with no delay
1     8  1701.369863   93701.369863                  46  Paid after 30 days
2     8  1380.821918   57380.821918                  28  Paid after 45 days
3     8  1109.589041   37109.589041                  18  Paid after 60 days
0     9          0.0      1628000.0                 814  Paid with no delay
1     9  1664.383562   91664.383562                  45  Paid after 30 days
2     9  1331.506849   55331.506849                  27  Paid after 45 days
3     9  1109.589041   37109.589041                  18  Paid after 60 days
0    10          0.0      1594000.0                 797  Paid with no delay
1    10   1627.39726    89627.39726                  44  Paid after 30 days
2    10  1331.506849   55331.506849                  27  Paid after 45 days
3    10  1109.589041   37109.589041                  18  Paid after 60 days
0    11          0.0      1562000.0                 781  Paid with no delay
1    11  1590.410959   87590.410959                  43  Paid after 30 days
2    11  1282.191781   53282.191781                  26  Paid after 45 days
3    11  1047.945205   35047.945205                  17  Paid after 60 days
0    12          0.0      1532000.0                 766  Paid with no delay
1    12  1590.410959   87590.410959                  43  Paid after 30 days
2    12  1282.191781   53282.191781                  26  Paid after 45 days
3    12  1047.945205   35047.945205                  17  Paid after 60 days

In [16]:
#Calculation of Interest, profit and loss:--------
interest_new=sum(data_new1['Interest'])
interest_old=sum(data_new['Interest'])
print("New Interest: ",interest_new)
print("Old Interest: ",interest_old)

#Profit calculation
profit_new=1000*(mem_fee + affl_card) + sum(data_new1['Interest'])
print("New Profit: ",profit_new)
print("Old Profit: ",profit)

#Liability calculation
liability_new=1000*((2000*cof_r) + opr_cost + 2000*loss_rate) #Assuming the loss rate on the outstanding balance default.
print("New Liability: ",liability_new)
print("Old Liability: ",liability)

New Interest:  37910.95890410959
Old Interest:  18955.479452054795
New Profit:  397910.9589041096
Old Profit:  378955.4794520548
New Liability:  490000.0
Old Liability:  395000.0


* As we can see that the Interest becomes double that of the previous interest value.
* There is a rise in profit approximately by $18,955 ($3,97,910 - $3,78,955).
* The liability increases by approximately by $95,000 (New Libility($4,90,000) - old Liability($3,95,000))

**3. Is the borrower with a low balance more or less likely to default then a borrower with a high balance? Why? Explain clearly.**

A survey report considering 2% sample of US consumers from three national credit bureaus showed that: 
* Most defaults occur on lower balances. There is a 32% default rate for balances of $5,000 or less and a 15% default rate for balances of more than $35,000. However, the majority of borrowers fall into the low balance category.

In my opinion, given real time data one can find the pattern of the defaulters list and reach conclusions based on that. 

**4. The group company has now become unhappy with the credit card provider. It wants to start its own credit card. But for that it will need to purchase the customers from the current credit card comapny. If the affiliated group offered Rs. 20 per account to purchase all customers from the credit card company, what would their profit be? (assume that we are only talking about 1 year)**

* If the affiliated company wants to buy all the customers of the credit-card company, then the profit of the credit-card company is given by:

In [17]:
purchase_cost=20
profit1=1000*(mem_fee + affl_card + purchase_cost) + sum(data_new['Interest']) 
profit=1000*(mem_fee + affl_card) + sum(data_new['Interest'])
print("New Profit:",profit1,"\nOld Profit:",profit)

New Profit: 398955.4794520548 
Old Profit: 378955.4794520548


* As we can see that there has been a slight increase in the profit of the company approximately by $20,000.

**5. What numbers can the financial institution change to convince the affiliated group not to want to purchase, while still going into a loss? Give the range of what new numbers are(min and max)**

**6. Is it beneficial for the company if the customer pays the credit card bill after the due date?(express your views as a buisness strategy analyst with a short example)**

* Disregarding the affiliation and the membership cost which are constants, the only way the credit card company can increase their profit margins is with the Interests collected due to payment after due date. Hence, it would be beneficial for the company if the customer pays credit-card bill after the due date.

Conclusions:
1. A late fee(charges) can be applied to the customers who are paying credit-card bill after the credit-free period.
2. Based on my assumptions, 10% of the customers are paying bill after the credit-free period, which restricts the company from making any profit. Among that 10%, 2% of the customers are revoked their credit-card services because of their late bill payement.